### PR TITLE
feat(mobile): Add to album from asset detail view

### PR DESCRIPTION
**Pull request by martyfuhry**
Works on https://github.com/immich-app/immich/issues/708.

It isn't strictly working, and I'm missing something on how to add to albums and update the album item count, but hopefully someone can tell me it's something obvious. I don't really understand the nature of remote versus local assets, so maybe that's it?

I also didn't add any of my strings to the localization strings because I wasn't sure if there was a process for it or anything.

I've also added a "New album" option in the "Add to album" options.

This, unfortunately, doesn't address adding an asset to a album that is shared with you, but I'm happy to work on that in this PR, as well.

image

### CODE REVIEW

File: mobile/lib/modules/album/views/create_album_page.dart [hunk 1]
```diff
@@ -26,7 +32,8 @@ class CreateAlbumPage extends HookConsumerWidget {
     final isAlbumTitleTextFieldFocus = useState(false);
     final isAlbumTitleEmpty = useState(true);
     final selectedAssets =
-        ref.watch(assetSelectionProvider).selectedNewAssetsForAlbum;
+        ref.watch(assetSelectionProvider).selectedNewAssetsForAlbum
+        ..addAll(initialAssets ?? []);
```
Comment (martyfuhry) [inline mobile/lib/modules/album/views/create_album_page.dart:36]: This doesn't work quite the way I intended it to. Got any suggestions for how to implement this idea?

**Comment by alextran1502** [comment]
Hello, thank you for the PR I hope I can help answer some questions that you have

```
Correctly adding the assets to the albums 
(it seems like it sometimes works, and sometimes doesn't, 
and I suspect it's got to do with the remote vs local nature of the assets).
```

Only remote asset (asset that is uploaded should be presented with the option to add to the album). For the actual implementation, please take a look at https://github.com/martyfuhry/immich/blob/6207b42dea1b5db38d0820a842ddd101144efbf7/mobile/lib/modules/home/views/home_page.dart#L98

I think that block of code would help answering the question on how to add assets to the album as well. Otherwise, let me know!

**Comment by martyfuhry** [comment]
That was extremely helpful and indeed fixed both of my issues. I still am having weird behavior with the `initialAssets` in the `create_album_page` where they are not being cleared between attempts. That is, if you select "Create new album", it passes the `initialAssets` as a single image and it seems to work OK. But subsequent attempts to "Create new album" with new assets seem to include the already included assets from the previous attempt... Odd. I'm sure I just don't understand how to properly clear the `Provider` cache or something.

Other than that, I'm very happy with how this is looking and feeling. I also suspect that some of the odd hit testing behavior is related to the `opaque` hit testing that should be present on the `GestureDetector`s. I don't typically use those when I can avoid it since they require more manual setup than, say, using an `InkWell` with a `Material`.

**Comment by martyfuhry** [comment]
I think that should fix the initial asset creation mechanism for a new album. I needed to use the `assetSelectionProvider` directly. Neat!

Hope I did any of this right!

File: mobile/lib/modules/album/views/create_album_page.dart (hunk 1 above)
Comment (martyfuhry) [reply mobile/lib/modules/album/views/create_album_page.dart]: I think I've resolved it below.

**Comment by alextran1502** [comment]
I've done manual testing and I noticed a few things

1. If you have both a non-shared album and shared albums created, on the interface of selecting an album to add the asset to, if you are the owner, you see all the albums including the shared album outside of the dropdown accordion.
2. The album doesn't seem to be updated when you click on the + button. I tried to change the thumbnail of the shared album on the web and it doesn't show the change on the mobile app in the selection bottom sheet
3. Maybe if there is no shared album existing, we don't show the accordion dropdown?

**Comment by martyfuhry** [comment]
Thank you so much for the feedback!

1. The shared album indeed shows up twice. I actually found the same behavior in the Google Photos app, so I suppose either their UX team thought it appropriate or else the de-duplication was too onerous on their system. I can suppress the shared albums from the "album" list, or else not show the "Shared option" if there are no albums being shared with you which you are not the owner. Alternatively, I can leave the behavior as is (matching Google Photos), if that's what is desired. I couldn't figure out where to check for the current user's owner ID, though. How do you access that?
2. I must have to update with one of the album description services or something. I'm running `ref.watch` on the `albumProvider`, `albumServiceProvider`, and the `sharedAlbumProvider`. Which additional provider do I need to watch? 
3. I agree. The `ExpansionTile` should be suppressed when the `sharedAlbum`s list is empty. This ties into (1) with the UX decision on duplicating / de-duplicating shared albums. If they are to be de-duplicated, should they be removed from the shared expansion tile or the main list?

**Comment by alextran1502** [comment]
If we are going to match Google Photos UI, I think they show another bottom sheet that contains all of the shared albums. This is what I see on iPhone, can you confirm you are seeing the same thing on Android? I think it makes sense to do it that way because you have the isolation level between the initial bottom sheet that shows the combination of non-shared and shared, then another view layer just shows the shared albums

the `watch` is for accessing the provider. So I think you have to do `ref.watch(albumProvider.notifier).getAllAlbums()` so that the provider would fetch new album data and then update the state.

**Comment by martyfuhry** [comment]
Maybe the Google Photos UI is different between Android & iOS.

In Android, I see this:
image

The above image is the "Add to album" bottom modal sheet. The "Shared" Expansion Tile is collapsed. You can see that the rest of my albums (shared and otherwise) show up below.

image

In the above image, I have now expanded the originally collapsed "Shared" expansion tile. You can see, e.g., the "Bathroom Floors" album show up in this list. I am the owner of it, and I'm sharing it with others. I can confirm it shows up twice for me, again below after I scroll through all of the shared albums.

I'm not advocating (necessarily) to do it the same way. I think it's kind of confusing, especially with the toy example you showed with only two albums.

How's it look on iOS?

**Comment by alextran1502** [comment]
Are you accessing Google Photos from its Web app?

Here is how it looks on iOS native app


https://user-images.githubusercontent.com/27055614/214700836-46f98df6-c944-4d1c-a802-e1ddba84a328.MP4

**Comment by martyfuhry** [comment]
Yeah my screenshots were from the Android app, not the web. 

Wow, the iOS native app uses stacked modal bottom sheets... That seems excessive. I definitely prefer the expandable tile here, but I'm very happy to change the design.

I was able to get the albums to update properly with the new cover images. I noticed a new bug in the library where deleting an image in the album doesn't update the album cover image or item count. I can open a new bug for that.

**Pull request merged**